commands/application.angular.js

```diff
@@ -17,34 +20,81 @@ function runSchematicCommand(schematicCommand, options, evaluatingOptions) {
         additionalOptions.push(schematicOption);
     };
 
-    const commandArguments = [
-        '-p', '@angular/cli@latest',
-        '-p', collectionPath,
+
+    let commandArguments = [
         'ng', 'g', `${collectionName}:${schematicCommand}`
     ].concat(additionalOptions);
 
-    runNpxCommand(commandArguments, evaluatingOptions);
+    const packageName = getNotInstalledLocalPackage(collectionName, evaluatingOptions && evaluatingOptions.cwd);
+
+    checkAngularPackage().then((result) => {
+        if(result) {
+            commandArguments = result.concat(commandArguments);
+        }
+
+        if(packageName) {
+            runCommand('npm', ['install', collectionName]).then(() => {
+                runCommand('npx', commandArguments, evaluatingOptions);
+            });
+        } else {
+            runCommand('npx', commandArguments, evaluatingOptions);
+        }
+    });
+}
+
+function getNotInstalledLocalPackage(packageName, cwd) {
+    if(fs.existsSync('node_modules') || cwd) {
+        const packageJsonPath = path.join(cwd || process.cwd(), `node_modules/${packageName}/package.json`);
+
+        if(!fs.existsSync(packageJsonPath)) {
+            return packageName;
+        }
+    }
+    return;
+}
+
+function checkAngularPackage() {
```

Comment:
This method is messy too. It named 'check...', so I expected the boolean result. But it returns the promise which could be resolved with a result or without. And the result is an array of additional arguments for the CLI command!